playwright/tests/pageTypes/pageTypesTable.spec.ts

```diff
@@ -0,0 +1,129 @@
+import { expect } from "@playwright/test";
+
+import { test } from "../../fixtures";
+import { generateRandomId } from "../../utils/generateRandomId";
+
+test.describe("Page types table", () => {
+  test.run()(
+    "I can create a reusable page type",
+    async ({ pageTypesTablePage, pageTypesBuilderPage }) => {
+      await pageTypesTablePage.goto();
```

Comment:
`.goto()` seems to be called at the beginning of every test. Should we use [`beforeEach()`](https://playwright.dev/docs/api/class-test#test-before-each-1) to reduce the boilerplate?

Reply:
I almost never use `beforeEach`, I know it's for that but also create a bit more complexity when you read tests. For me, a good pattern is always to be able to read and understand a test without having to remember some part that is done before for you somewhere else in the file. I feel the duplication is not a big problem in tests, and I tend to duplicate tests over adding more complexity. At the contrary of what I would do in a prod code where duplication of code is prone to bugs, of course.

My comment here can also refer to the comment above, as I tend to duplicate to simplify code most of the time.

If you can quickly read [this article](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing) you will see what kind of dev I'm for test haha. Mostly I'm talking about the [Nesting](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing#nesting) part. I really like Kent C. Dodds on testing subject and I mostly agree with his articles.

I think for the comment above, it makes more sense because we are actually describing enough what the check behind do with the function name, and it's our pattern with POM, so I like it and will do that 🙂

I'm more afraid of bringing a `beforeEach` here because I would like us to write tests in a `flat` manner, where everything can be understood from a single test. 

WDYT?

Reply:
I totally agree about being able to understand the whole test within the test's function. Your reasoning to keep `beforeEach()` out of the tests makes sense to me. 👍

Same for the field name and ID checks. I prefer duplication as well over premature abstraction. It's a balance, and I think we're still on the positive side. 🙂